frontend/lib/src/components/shared/Toolbar/Toolbar.tsx

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactElement } from "react"
+
+import { EmotionIcon } from "@emotion-icons/emotion-icon"
+import { useTheme } from "@emotion/react"
+import { StyledComponent } from "@emotion/styled"
+import { Fullscreen, FullscreenExit } from "@emotion-icons/material-outlined"
+
+import StreamlitMarkdown from "@streamlit/lib/src/components/shared/StreamlitMarkdown"
+import Tooltip, {
+  Placement,
+} from "@streamlit/lib/src/components/shared/Tooltip"
+import Button, {
+  BaseButtonKind,
+} from "@streamlit/lib/src/components/shared/BaseButton"
+import Icon from "@streamlit/lib/src/components/shared/Icon"
+import { LibContext } from "@streamlit/lib/src/components/core/LibContext"
+import { EmotionTheme } from "@streamlit/lib/src/theme"
+
+import { StyledToolbar, StyledToolbarWrapper } from "./styled-components"
+
+export interface ToolbarActionProps {
+  label: string
+  icon?: EmotionIcon
+  show_label?: boolean
+  onClick: () => void
+}
+
+export function ToolbarAction({
+  label,
+  show_label,
+  icon,
+  onClick,
+}: ToolbarActionProps): ReactElement {
+  const theme: EmotionTheme = useTheme()
+
+  const displayLabel = show_label ? label : ""
+  return (
+    <div data-testid="stElementToolbarButton">
+      <Tooltip
+        content={
+          <StreamlitMarkdown
+            source={label}
+            allowHTML={false}
+            style={{ fontSize: theme.fontSizes.sm }}
+          />
+        }
+        placement={Placement.TOP}
+        onMouseEnterDelay={1000}
```

Comment:
What is this delay for?

Reply:
The default tooltip delay that we use in Streamlit is a bit too fast in the case of this toolbar. Therefore, we are applying a higher delay (1s) here. I will add a comment.

Reply:
Added a comment 👍